### PR TITLE
Refactor AI capturing.

### PR DESCRIPTION
**Pull request by GraionDilach**

### CODE REVIEW

File: OpenRA.Mods.HV/Traits/BotModules/CaptureManagerHVBotModule.cs [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using OpenRA.Mods.Common;
+using OpenRA.Mods.Common.Traits;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.HV.Traits
+{
+	[Desc("Manages AI capturing logic.")]
+	public class CaptureManagerHVBotModuleInfo : ConditionalTraitInfo
```
Comment (Mailaender) [inline OpenRA.Mods.HV/Traits/BotModules/CaptureManagerHVBotModule.cs:22]: Maybe
```suggestion
	public class PriorityCaptureManagerBotModuleInfo : ConditionalTraitInfo
```

**Comment by GraionDilach** [comment]
Applied the suggested name.

File: OpenRA.Mods.HV/Traits/BotModules/PriorityCaptureManagerBotModule.cs [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+
+							capturers = capturers.Skip(1);
+						}
+
+						priorityTargets = priorityTargets.Skip(1);
+					}
+				}
+
+				if (!capturers.Any())
+					return;
```
Comment (Mailaender) [inline OpenRA.Mods.HV/Traits/BotModules/PriorityCaptureManagerBotModule.cs:184]: This is duplicated and probably wrong here.

File: OpenRA.Mods.HV/Traits/BotModules/PriorityCaptureManagerBotModule.cs [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+			}
+
+			var randPlayer = world.Players.Where(p => !p.Spectating
```
Comment (Mailaender) [inline OpenRA.Mods.HV/Traits/BotModules/PriorityCaptureManagerBotModule.cs:187]:
```suggestion
			var randomPlayer = world.Players.Where(p => !p.Spectating
```

File: OpenRA.Mods.HV/Traits/BotModules/PriorityCaptureManagerBotModule.cs (hunk 2 above)
Comment (GraionDilach) [reply OpenRA.Mods.HV/Traits/BotModules/PriorityCaptureManagerBotModule.cs:184]: It goes to a line above, true that. This is an early return incase all capturers were used up for priority.

File: OpenRA.Mods.HV/Traits/BotModules/PriorityCaptureManagerBotModule.cs [hunk 4]
```diff
@@ -0,0 +1,245 @@
...
+						var priorityTarget = priorityTargets.First();
+
+						var captureManager = priorityTarget.TraitOrDefault<CaptureManager>();
+						if (captureManager != null && captureManager.CanBeTargetedBy(priorityTarget, capturer.Actor, capturer.Trait))
+						{
+							bot.QueueOrder(new Order("CaptureActor", capturer.Actor, Target.FromActor(priorityTarget), true));
+							AIUtils.BotDebug("AI ({0}): Ordered {1} {2} to capture {3} {4} in priority mode.",
+								player.ClientIndex, capturer.Actor, capturer.Actor.ActorID, priorityTarget, priorityTarget.ActorID);
+
+							capturers = capturers.Skip(1);
```
Comment (GraionDilach) [inline OpenRA.Mods.HV/Traits/BotModules/PriorityCaptureManagerBotModule.cs:176]: This should skip even if this block fails.

**Pull request merged**